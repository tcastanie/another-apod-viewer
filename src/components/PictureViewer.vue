<template>
  <LoadingLayer v-if="!imgLoaded" />
  <div class="c-picture-viewer">
    <Transition>
      <a v-show="imgLoaded" class="c-picture-viewer__img" :href="fullUrl">
        <img :src="imgUrl" @load="onImgLoad" />
      </a>
    </Transition>
  </div>
</template>

<script>
import LoadingLayer from "./LoadingLayer.vue";
export default {
  components: {
    LoadingLayer,
  },
  props: {
    imgUrl: {
      type: String,
      default: "",
    },
    fullUrl: {
      type: String,
      default: "",
    },
  },
  data() {
    return {
      imgLoaded: false,
    };
  },
  methods: {
    onImgLoad() {
      this.imgLoaded = true;
    },
  },
};
</script>

<style lang="scss" scoped>
.c-picture-viewer {
  display: flex;
  align-items: center;
  justify-content: center;

  &__img {
    img {
      max-width: 100%;
      height: auto;
      max-height: var(--max-picture-height);
    }
  }
}

.v-enter-active,
.v-leave-active {
  transition: all 0.5s ease;
}

.v-enter-from,
.v-leave-to {
  opacity: 0;
}
</style>
