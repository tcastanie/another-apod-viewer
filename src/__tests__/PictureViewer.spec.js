import { describe, it, expect } from "vitest";
import { mount } from "@vue/test-utils";
import PictureViewer from "@/components/PictureViewer.vue";
import { imgUrl, fullUrl } from "./constantsMockups";

describe("PictureViewer", () => {
  it("renders properly", () => {
    const wrapper = mount(PictureViewer, {
      props: {
        imgUrl,
        fullUrl,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("display image correctly ", () => {
    const wrapper = mount(PictureViewer, {
      props: {
        imgUrl,
        fullUrl,
      },
    });
    expect(wrapper.vm.imgLoaded).toBe(false);
    wrapper.find("img").trigger("load");
    expect(wrapper.vm.imgLoaded).toBe(true);
  });
});
