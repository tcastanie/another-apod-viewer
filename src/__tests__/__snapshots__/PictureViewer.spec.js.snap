// Vitest Snapshot v1

exports[`PictureViewer > renders properly 1`] = `
"<article aria-busy=\\"true\\" class=\\"c-loading\\" data-v-4320971b=\\"\\" data-v-eea99233=\\"\\">Loading...</article>
<div class=\\"c-picture-viewer\\" data-v-eea99233=\\"\\">
  <transition-stub data-v-eea99233=\\"\\"><a class=\\"c-picture-viewer__img\\" href=\\"https://apod.nasa.gov/apod/image/2207/SaturnIR_CassiniKakitsev_1024.jpg\\" target=\\"_blank\\" data-v-eea99233=\\"\\" style=\\"display: none;\\"><img src=\\"https://apod.nasa.gov/apod/image/2207/SaturnIR_CassiniKakitsev_960.jpg\\" data-v-eea99233=\\"\\"></a></transition-stub>
</div>"
`;
